win_etw_macros/src/lib.rs

```diff
@@ -395,20 +400,47 @@ fn trace_logging_events_core(attr: TokenStream, item_tokens: TokenStream) -> Tok
                 method.span(),
                 "Event metadata is too large to encode; reduce the complexity of this event.",
             ));
-            continue;
+            break;
         }
         let event_metadata_len_b0 = (event_metadata_len & 0xff) as u8;
         let event_metadata_len_b1 = (event_metadata_len >> 8) as u8;
         event_metadata[0] = parse_quote! { #event_metadata_len_b0 };
         event_metadata[1] = parse_quote! { #event_metadata_len_b1 };
 
-        let event_attrs = parse_event_attributes(&mut errors, &method.sig.ident, &method.attrs);
+        let event_attrs = parse_event_attributes(&method.sig.ident, &method.attrs);
+
+        if let Err(e) = event_attrs {
+            errors.push(e);
+            break;
+        }
+
+        let event_attrs = event_attrs.unwrap();
 
         // Generate the event descriptor for this event.
         // This is a static variable. The name is exactly the name of the event.
         let event_level = event_attrs.level;
         let event_opcode = event_attrs.opcode;
         let event_task = event_attrs.task;
+        let event_id = event_attrs.event_id;
+
+        // Set this on first iteration, check on all others.
+        if event_idx == 0 {
+            auto_generated = event_id.is_none();
+        } else if auto_generated == event_id.is_some() {
+            errors.push(Error::new(
+                method.span(),
+                "Event ids must be set for all events, or for none.",
+            ));
+            break;
+        }
+
+        let event_id = event_id.unwrap_or(event_idx);
+
+        if !event_ids.insert(event_id) {
```

Comment:
For good diagnostics, could you please report both of the conflicting events?  Something like this:

```
if let Some(conflicting_event) = event_ids.get(&event_id) {
  errors.push(Error::new(method.span(), format!("The event id {} has a conflict. It has been assigned to both the '{}' and '{}' events.",
    event_id,
    conflicting_event.name,
    method.name));
}
```

This really helps save time, when people are dealing with errors. There's even a way to report an error associated with the span of the other side of the conflict, but I don't recall what it is right now.

Reply:
The numbers are the same so this does not really help :(

Reply:
Let me see if I can save the name of the function in a hashset or something with the event id.